localstack/services/stepfunctions/provider_v2.py

```diff
@@ -147,6 +158,46 @@ def describe_state_machine(
             loggingConfiguration=sm.logging_config,
         )
 
+    def send_task_success(
+        self, context: RequestContext, task_token: TaskToken, output: SensitiveData
+    ) -> SendTaskSuccessOutput:
+        outcome = CallbackOutcomeSuccess(callback_id=task_token, output=output)
+        store = self.get_store(context)
+        for exec in store.executions.values():
+            try:
+                if exec.exec_worker.env.callback_pool_manager.notify(
+                    callback_id=task_token, outcome=outcome
+                ):
+                    return SendTaskSuccessOutput()
```

Comment:
the access pattern here seems a bit noisy/inefficient going through all executions but we can iterate over that at a later point :+1:

Reply:
Yes I definitely agree. I'll not it down and pick it up after this stream of prs. This should make things easier as there is a refactoring of this strategy in one of these (eg: https://github.com/localstack/localstack/pull/8428/files#diff-a5c8a2c872564256e1186b4766cd307a568332bb5f0dd7f709f39f24ad645e4aL16) :)